DataFormats/EgammaCandidates/interface/HIPhotonIsolation.h

```diff
@@ -0,0 +1,125 @@
+#ifndef EgammaCandidates_HIPhotonIsolation_h
+#define EgammaCandidates_HIPhotonIsolation_h
+
+#include "DataFormats/Common/interface/ValueMap.h"
+
+namespace reco{
+
+  class HIPhotonIsolation {
+
+  public:
+
+  HIPhotonIsolation() : ecalClusterIsoR1_(0),
+      ecalClusterIsoR2_(0),
+      ecalClusterIsoR3_(0),
+      ecalClusterIsoR4_(0),
+      ecalClusterIsoR5_(0),
+      hcalRechitIsoR1_(0),
+      hcalRechitIsoR2_(0),
+      hcalRechitIsoR3_(0),
+      hcalRechitIsoR4_(0),
+      hcalRechitIsoR5_(0),
+      trackIsoR1PtCut20_(0),
+      trackIsoR2PtCut20_(0),
+      trackIsoR3PtCut20_(0),
+      trackIsoR4PtCut20_(0),
+      trackIsoR5PtCut20_(0),
+      swissCrx_(0),
+      seedTime_(0)
+      {}
+    virtual ~HIPhotonIsolation() {}
+
+    //getters
+
+    /// Cluster-based isolation (ECAL) R = 0.1
+    float ecalClusterIsoR1() const {return ecalClusterIsoR1_;}
+    /// Cluster-based isolation (ECAL) R = 0.2
+    float ecalClusterIsoR2() const {return ecalClusterIsoR2_;}
+    /// Cluster-based isolation (ECAL) R = 0.3
+    float ecalClusterIsoR3() const {return ecalClusterIsoR3_;}
+    /// Cluster-based isolation (ECAL) R = 0.4
+    float ecalClusterIsoR4() const {return ecalClusterIsoR4_;}
+    /// Cluster-based isolation (ECAL) R = 0.5
+    float ecalClusterIsoR5() const {return ecalClusterIsoR5_;}
+
+    /// Rechit-based isolation (HCAL) R = 0.1
+    float hcalRechitIsoR1() const {return hcalRechitIsoR1_;}
+    /// Rechit-based isolation (HCAL) R = 0.2
+    float hcalRechitIsoR2() const {return hcalRechitIsoR2_;}
+    /// Rechit-based isolation (HCAL) R = 0.3
+    float hcalRechitIsoR3() const {return hcalRechitIsoR3_;}
+    /// Rechit-based isolation (HCAL) R = 0.4
+    float hcalRechitIsoR4() const {return hcalRechitIsoR4_;}
+    /// Rechit-based isolation (HCAL) R = 0.5
+    float hcalRechitIsoR5() const {return hcalRechitIsoR5_;}
+
+    /// Track-based isolation, pt>2.0GeV, R = 0.1
+    float trackIsoR1PtCut20() const {return trackIsoR1PtCut20_;}
+    /// Track-based isolation, pt>2.0GeV, R = 0.2
+    float trackIsoR2PtCut20() const {return trackIsoR2PtCut20_;}
+    /// Track-based isolation, pt>2.0GeV, R = 0.3
+    float trackIsoR3PtCut20() const {return trackIsoR3PtCut20_;}
+    /// Track-based isolation, pt>2.0GeV, R = 0.4
+    float trackIsoR4PtCut20() const {return trackIsoR4PtCut20_;}
+    /// Track-based isolation, pt>2.0GeV, R = 0.5
+    float trackIsoR5PtCut20() const {return trackIsoR5PtCut20_;}
+
+    /// SwissCross crystal ratio
+    float swissCrx() const {return swissCrx_;}
+    /// Ecal rechit seed time
+    float seedTime() const {return seedTime_;}
+
+    // setters
+
+    /// Cluster-based isolation (ECAL) R = 0.1
+    void ecalClusterIsoR1(float ecalClusterIsoR1)  {ecalClusterIsoR1_ = ecalClusterIsoR1;}
```

Comment:
it would be more appropriate to have the modifier methods have a different name.
Identical names but different signature in the argument list still works though.